### PR TITLE
Feat: Trunk based deployment flow

### PR DESCRIPTION
**Pull request by invke**
### 📝 Description

This will enable continuous deployment and management of the environments when accompanied with automated re-tagging as develop is updated. The aim is to remove the frustration caused by using the dev/qa branches and make our internal environments a representation of the trunk by default.

You can see how this is being used with the first implementation using it on BPPlus:

- https://github.com/paperkite/bpau-bpme-api/pull/309

#### Automated deployment usage

Here's an example automated deployment workflow:

- https://github.com/paperkite/bpau-bpme-api/actions/runs/3464869416

image

#### Automated retagging usage

I have tested the automated retagging when it needs to retag, but I can't find my workflow run. I can reverify it but it's a little bit of playing around to do so without it merged. Here's an example of the automated retagging wasn't needed (it's not in history):

- https://github.com/paperkite/bpau-bpme-api/actions/runs/3464860395

image

### CODE REVIEW

**Review by deep12650 — APPROVED** [review]

File: generate-environment-info/action.yml
Comment (invke) [inline generate-environment-info/action.yml:68]: I guess I deserved this, merging without waiting for second approval. Will fix in follow up PR.

**Pull request merged**
